src/Microsoft.Azure.WebJobs.Host/Executors/FunctionExecutor.cs

```diff
@@ -499,11 +521,22 @@ private static ITaskSeriesTimer StartParameterLogTimer(IRecurrentCommand updateC
                 await singleton.AcquireAsync(functionCancellationTokenSource.Token);
             }
 
-            var timer = StartFunctionTimeout(instance, globalFunctionTimeout, functionCancellationTokenSource, traceWriter);
+            var timer = StartFunctionTimeout(instance, globalFunctionTimeout, timeoutWhileDebugging, functionCancellationTokenSource, traceWriter);
             try
             {
                 await invoker.InvokeAsync(invokeParameters);
             }
+            catch (OperationCanceledException)
+            {
+                // if cancellationToken is not signaled, it means this is a timeout
+                if (throwOnFunctionTimeout && !cancellationToken.IsCancellationRequested)
```

Comment:
Right, we need to throw here so the rest of the logging, etc, happens before we rethrow the 'fatal' exception. Having the timer delegate throw means the exception is coming from some other thread.

Good idea on the additional delegate. I'll do that.

---
In reply to: [76653248](https://github.com/Azure/azure-webjobs-sdk/pull/799#discussion_r76653248) [](ancestors = 76653248)